tests/neg-custom-args/erased/i4060.scala

```diff
@@ -0,0 +1,22 @@
+// See https://github.com/lampepfl/dotty/issues/4060#issuecomment-445808377
+import scala.language.experimental.erasedDefinitions
+
+object App {
+  trait A { type L >: Any}
+  def upcast(erased a: A)(x: Any): a.L = x
+  erased val p: A { type L <: Nothing } = p
+  def coerce(x: Any): Int = upcast(p)(x) // error
+
+  def coerceInline(x: Any): Int = upcast(compiletime.erasedValue[A {type L <: Nothing}])(x) // error
+
+  trait B { type L <: Nothing }
+  def upcast_dep_parameter(erased a: B)(x: a.L) : Int = x
+  erased val q : B { type L >: Any } = compiletime.erasedValue
```

Comment:
Why is no error emitted on this line? What happens if I try to use `q`'s bounds directly on the line below, without passing it as an argument to any function?

I feel like realizability should be checked whenever we try to create an erased value.

Reply:
Realizability checks are always performed when bounds are used, so using `q`'s bounds directly will CE even without this PR.
This PR only adds the checks at function calls (they were performed at definition before, but they are not sufficient per the example).

I had a chat with @odersky about whether checks should be done when we create erased values. We concluded that it's okay to leave it at that to maximize flexibility of erased values.

Reply:
Ah. Then everything looks good to me.